packages/vm/src/cache.rs

```diff
@@ -16,6 +16,9 @@ use crate::wasm_backend::{compile_and_use, compile_only, make_runtime_store};
 
 const WASM_DIR: &str = "wasm";
 const MODULES_DIR: &str = "modules";
+// Rounded-up average of Module size to optimized wasm size ratio
+// Based on `examples/module_size.sh`, and the (optimized) cosmwasm-plus modules
+const OPTIMIZED_WASM_SIZE_FACTOR: usize = 18;
```

Comment:
I'd add a little safety margin. What about 20?

Reply:
OK. I've used 18 because the "average" (least squares fit) of the cosmwasm-plus modules is 17.5762. And, the linear approximation is pretty good.

Reply:
It would be great to document that and just say: we observed up to 17.5762 over a variety of contracts but need to assume an attacker was able to find a higher factor, so we choose 20.

Reply:
Related: is there a way to know if a contract (wasm file contents) is optimized or not?

Not really a problem, but for non-optimized contracts, this factor is way higher.

Reply:
For non-optimized the factor is way smaller. When the Wasm bytes increase without adding much logic, the Module size increases less. So Wasm and Module get closer to each other.

Reply:
> So Wasm and Module get closer to each other.

Yes. But then, if we multiply a non-optimized wasm size by this factor, we'll be overestimating the Module size, by a great deal (20 / ~4 = ~5).

Reply:
Jupp, but that's okay I think. Preventing memory attacks is more important that ensuring the memory limit is fully used.